cmd/zoekt-sourcegraph-indexserver/main_test.go

```diff
@@ -90,6 +85,30 @@ func TestListRepoIDs(t *testing.T) {
 	}
 }
 
+func TestListRepoIDs_Error(t *testing.T) {
+	msg := "deadbeaf deadbeaf"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		// This is how Sourcegraph returns error messages to the caller.
+		http.Error(w, msg, http.StatusInternalServerError)
```

Comment:
did you test this end to end just to make sure we do indeed send StatusInternalServerError?

Reply:
yes, I returned an err from serveList in Sourcegraph and saw the message in indexserver's logs.